File: Day01/Solution.py
###Advent of Code###
with open('Day01\input.txt') as f:
    lines = [int(x) for x in f.read().split()]

# task 1
count = 0
for ind, val in enumerate(lines):
    if ind != 0 or ind != len(lines) - 1:
        if val > lines[ind - 1]:
            count += 1

print('Task 1:', count)

# task 2
count = 0
for ind, val in enumerate(lines):
    if ind < len(lines) - 3:
        A = val + lines[ind + 1] + lines[ind + 2]
        B = lines[ind + 1] + lines[ind + 2] + lines[ind + 3]
        if B > A:
            count += 1

print('Task 2:', count)

File: Day01/Solution.jl
#=
Solution:
- Julia version: 1.7.0
- Author: Jora Singh Randhawa
- Date: 2021-12-01
=#

input = open("Day01/input.txt", "r")
depths = parse.(Int, readlines(input))

# Task 1
function looping(depth)
    count = 0
    for (ind, val) in enumerate(depths)
        if ind != 1
            if val > depths[ind-1]
                count += 1
            end
        end
    end
    println("Task 1: ", count)
end
looping(depths)


# Task 2
function looping(depth)
    count = 0
    for (ind, val) in enumerate(depths)
        if ind < (size(depths, 1) - 2)
            A = val + depths[ind + 1] + depths[ind + 2]
            B = depths[ind + 1] + depths[ind + 2] + depths[ind + 3]
            if B > A
                count += 1
            end
        end
    end
    println("Task 2: ", count)
end
looping(depths)

